### ISSUE TITLE
v4 - refresh no longer working

### ISSUE THREAD

**Issue by bcorporaal** [cocopon/tweakpane#570]
I am trying to migrate code from Tweakpane v3 to v4. All is working well, except for the refresh function. The code using it that was working for v3 no longer works. Specifically, nothing happens.

Here is an example panel I'm working on:
<img width="323" alt="Screenshot 2023-09-03 at 17 48 08" src="https://github.com/cocopon/tweakpane/assets/12683014/a6e935c6-f3b6-4552-bffa-ddc5e3322af6">
In the Colors section, clicking the arrows or the rnd buttons changes the palette value and triggers a refresh. Previously then the palette dropdown would update and the visualization would redraw. With v4 nothing happens. No changes, no errors.

Digging into the Tweakpane code and adding a few log statements, it seems that in the refresh function (line 1708) the children array it tries to iterate only has 1 child and that has isRefreshable = false. Given that there are multiple inputs and taps in this instance, I would imagine it needs to iterate more than 1 child.

Again, this all worked with v3, so I'm not sure what is going on.

**Comment by cocopon** [comment]
This small test code works fine in my environment:

```ts
const PARAMS = {someProp: 1};
const pane = new Pane();

// Adding some bindings
pane.addBinding(PARAMS, 'someProp');
pane.addBinding(PARAMS, 'someProp', {options: {'foo': 1, 'bar': 2}});

// Update the value outside the pane
PARAMS.someProp = 2;

// Refresh the pane
pane.refresh();
// or, pane.children[i].refresh() also works fine
```

It seems that more information about your environment is needed to reproduce the problem. Sharing a code snippet or a minimal environment would be helpful.

**Comment by bcorporaal** [comment]
Thanks for our input. The `pane.children[i[.refresh()` comment got me thinking and I got it working. As you can see in the screenshot, I have multiple tabs. If instead of `pane.refresh()`, I use `pane.pages[0].refresh()`, it works for me.

Not sure if this is the intended behavior, but for me this works.

If you're curious, you can see it in my code for [Toko](https://github.com/bcorporaal/Toko/tree/v0.6), a library to extend p5.js. Here [line 29](https://github.com/bcorporaal/Toko/blob/v0.6/toko/source/core/tweakpane.js) and onwards, I set up the 3 buttons, and line 55 is where I had to change it. (`pane.tab` is a shortcut to `tweakpane.pages[0]`).

**Comment by cocopon** [comment]
Thank you for the details! I found the cause and created a new issue: #571

**Comment by cocopon** [comment]
Fixed in 4.0.1.
